diff --git a/libavcodec/avcodec.h b/libavcodec/avcodec.h
index 18c3e3e..1b98f5c 100644
--- a/libavcodec/avcodec.h
+++ b/libavcodec/avcodec.h
@@ -3702,6 +3702,13 @@ typedef struct AVCodecContext {
      * (with the display dimensions being determined by the crop_* fields).
      */
     int apply_cropping;
+
+    /**
+     * WHA player version.
+     * - encoding: unused
+     * - decoding: set by user
+     */
+    int wha_version;
 } AVCodecContext;
 
 AVRational av_codec_get_pkt_timebase         (const AVCodecContext *avctx);
diff --git a/libavcodec/options_table.h b/libavcodec/options_table.h
index 2ac37c3..bf5ec66 100644
--- a/libavcodec/options_table.h
+++ b/libavcodec/options_table.h
@@ -580,6 +580,7 @@ static const AVOption avcodec_options[] = {
 {"ignore_level", "ignore level even if the codec level used is unknown or higher than the maximum supported level reported by the hardware driver", 0, AV_OPT_TYPE_CONST, { .i64 = AV_HWACCEL_FLAG_IGNORE_LEVEL }, INT_MIN, INT_MAX, V | D, "hwaccel_flags" },
 {"allow_high_depth", "allow to output YUV pixel formats with a different chroma sampling than 4:2:0 and/or other than 8 bits per component", 0, AV_OPT_TYPE_CONST, {.i64 = AV_HWACCEL_FLAG_ALLOW_HIGH_DEPTH }, INT_MIN, INT_MAX, V | D, "hwaccel_flags"},
 {"allow_profile_mismatch", "attempt to decode anyway if HW accelerated decoder's supported profiles do not exactly match the stream", 0, AV_OPT_TYPE_CONST, {.i64 = AV_HWACCEL_FLAG_ALLOW_PROFILE_MISMATCH }, INT_MIN, INT_MAX, V | D, "hwaccel_flags"},
+{"wha_version", "WHA player version", OFFSET(wha_version), AV_OPT_TYPE_INT, {.i64 = WHA_VERSION_MAX}, WHA_VERSION_MIN, WHA_VERSION_MAX, D, "wha"},
 {NULL},
 };
 
diff --git a/libavformat/avformat.h b/libavformat/avformat.h
index b0de66a..edfef92 100644
--- a/libavformat/avformat.h
+++ b/libavformat/avformat.h
@@ -1917,6 +1917,13 @@ typedef struct AVFormatContext {
      * - decoding: set by user
      */
     int max_streams;
+
+    /**
+     * WHA player version.
+     * - encoding: unused
+     * - decoding: set by user
+     */
+    int wha_version;
 } AVFormatContext;
 
 /**
diff --git a/libavformat/options_table.h b/libavformat/options_table.h
index 118086d..cdb6d38 100644
--- a/libavformat/options_table.h
+++ b/libavformat/options_table.h
@@ -108,6 +108,7 @@ static const AVOption avformat_options[] = {
 {"protocol_whitelist", "List of protocols that are allowed to be used", OFFSET(protocol_whitelist), AV_OPT_TYPE_STRING, { .str = NULL },  CHAR_MIN, CHAR_MAX, D },
 {"protocol_blacklist", "List of protocols that are not allowed to be used", OFFSET(protocol_blacklist), AV_OPT_TYPE_STRING, { .str = NULL },  CHAR_MIN, CHAR_MAX, D },
 {"max_streams", "maximum number of streams", OFFSET(max_streams), AV_OPT_TYPE_INT, { .i64 = 1000 }, 0, INT_MAX, D },
+{"wha_version", "WHA player version", OFFSET(wha_version), AV_OPT_TYPE_INT, {.i64 = WHA_VERSION_MAX}, WHA_VERSION_MIN, WHA_VERSION_MAX, D, "wha"},
 {NULL},
 };
 
diff --git a/libavutil/version.h b/libavutil/version.h
index f594dc0..707669c 100644
--- a/libavutil/version.h
+++ b/libavutil/version.h
@@ -140,6 +140,8 @@
 #define FF_API_CRYPTO_SIZE_T            (LIBAVUTIL_VERSION_MAJOR < 56)
 #endif
 
+#define WHA_VERSION_MIN 0
+#define WHA_VERSION_MAX 1
 
 /**
  * @}
